kernel/src/allocator/mem_alloc.c

```diff
@@ -0,0 +1,87 @@
+#include <klibc.h>
```

Comment:
Maybe add comments to public methods?

Reply:
klibc needs to be updated in general. A bit out of the scope of this PR imo. Will be addressed soon

Reply:
@jonay2000 I think he is referring to `mem_alloc.c` not klibc

Reply:
hmm you're right. Those are basically private methods though. Though the comment at the top of mem_alloc.c is wrong